v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/types/hardfork.ts

```diff
@@ -43,15 +50,10 @@ export function getHardforks(chainType: ChainType): string[] {
     : L1_HARDFORK_ORDER;
 }
 
-const L1_LATEST_HARDFORK: L1HardforkName =
-  L1_HARDFORK_ORDER[L1_HARDFORK_ORDER.length - 1];
-const OP_LATEST_HARDFORK: OpHardforkName =
-  OP_HARDFORK_ORDER[OP_HARDFORK_ORDER.length - 1];
-
 export function getCurrentHardfork(chainType: ChainType): string {
   return chainType === OPTIMISM_CHAIN_TYPE
-    ? OP_LATEST_HARDFORK
-    : L1_LATEST_HARDFORK;
+    ? edrOpHardforkToHardhatOpHardforkName(opLatestHardfork())
```

Comment:
Instead of using the last hardfork in the list, we use EDR's reported "latest" hardfork.

Once the Osaka hardfork is activated in mainnet, EDR will do a new release to switch the "latest" hardfork to Osaka.